### PR TITLE
Want to refactor GetEventstoreRepository name and remove/change signatures

### PR DESCRIPTION
**Pull request by dvoaviarisonld**
This PR contains the following changes
1. This should be the EventStore repository as the GetEventStore domain was removed a coupe of years back.

2. The outbus parameter should be remove from the constructor. (this may impact a number of tests, break into a separate ticket if this becomes a significant refactor)

3. Allow injection of the stream name generator, including optional prefix ("Domain".), and standard suffix options (":Snapshot" ect.)
```csharp
(t, g) => string.Format($"{domainPrefix}.{char.ToLower(t.Name[0]) + t.Name.Substring(1)}-{g:N}" 
```
this should be wrapped in a well know type of some sort, rather than being a raw function.

4. Remove commitId and header update from the Save Method signature (commit id should still be generated and saved in the header)

5. Add the typename to the headers
```csharp
{EventClrQualifiedTypeHeader, aggregate.GetType().AssemblyQualifiedName},
{EventClrTypeHeader, aggregate.GetType().Name}
```
n.b. this will require updating the deserialization
this will support cross domain support for deserialization

For the future:
add a flag to the repo to allow fallback deserialization by just type name, but this should not be the default

(RD-20) WIP

### CODE REVIEW

File: src/ReactiveDomain.Foundation/EventStore/EventStoreRepository.cs [hunk 1]
```diff
@@ -122,6 +114,12 @@ private static TAggregate ConstructAggregate<TAggregate>()
             return (TAggregate)Activator.CreateInstance(typeof(TAggregate), true);
         }
 
+        private string GetStreamNameFor(Type type, Guid id)
+        {
+            // standard stream name formating [lowercaseprefix].[camelCaseName]-[id]
+            return $"{_streamNamePrefix.ToLower()}.{char.ToLower(type.Name[0])}{type.Name.Substring(1)}-{id:N}";
```
Comment (StephenDrewLDS) [inline src/ReactiveDomain.Foundation/EventStore/EventStoreRepository.cs]: `ToLowerInvariant()` ?

Comment (dvoaviarisonld) [reply src/ReactiveDomain.Foundation/EventStore/EventStoreRepository.cs]: Probably, yes

**Comment by condron** [comment]
Merged some command bus updates to start fixing the tests. I can't test locally because everything passes.

File: src/ReactiveDomain.Foundation/EventStore/EventStoreRepository.cs (hunk 1 above)
Comment (condron) [reply src/ReactiveDomain.Foundation/EventStore/EventStoreRepository.cs]: yup

**Comment by condron** [comment]
@dvoaviarisonld  We should create a new class StreamNameBuilder and pass that into the repository, it need to take the prefix into it's constructor. No prefix is ok, but must be explicit. In that case make sure the  '.' is not added to the name.

**Comment by condron** [comment]
removed the helper classes from the nested command tests

**Comment by dvoaviarisonld** [comment]
Will merge any changes related to RD-20 here (any sub-issues) so renamed the PR accordingly

**Pull request merged**